reframe/frontend/executors/policies.py

```diff
@@ -100,6 +101,14 @@ def runcase(self, case):
                    for c in case.deps if c in self._task_index):
                 raise TaskDependencyError('dependencies failed')
 
+            if any(self._task_index[c].skipped
+                   for c in case.deps if c in self._task_index):
+                try:
+                    raise SkipTestError('skipped due to skipped dependencies')
+                except SkipTestError as e:
+                    task.skip()
+                    raise TaskExit from e
```

Comment:
I don't understand this code; you are raising the `SkipTestError` and you are catching it immediately?

Reply:
Yes, because otherwise I would have to create the `exc_info` manually and pass it to `skip()`.  It's not a big deal, but otherwise I would also have to create a stack trace manually, which here I get it for free. I should add a comment to explain this.